app/src/components/AddCredentialSlider.tsx

```diff
@@ -79,39 +77,34 @@ const AddCredentialSlider: React.FC = () => {
     DeviceEventEmitter.emit(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, false)
   }, [])
 
-  const goToHomeScreen = (credentialId?: string) => {
-    navigation.getParent()?.navigate(Stacks.NotificationStack, {
-      screen: Screens.CredentialOffer,
-      params: { credentialId },
-    })
-  }
-
   const goToScanScreen = useCallback(() => {
     deactivateSlider()
     navigation.getParent()?.navigate(Stacks.ConnectStack, { screen: Screens.Scan })
   }, [])
 
   const onBCIDPress = useCallback(() => {
-    setWorkflowInFlight(true)
-    startFlow(agent!, store, setWorkflowInFlight, t, (connectionId) => setWorkflowConnectionId(connectionId))
+    setWorkflowInProgress(true)
+    startFlow(agent!, store, setWorkflowInProgress, t, (connectionId) => setWorkflowConnectionId(connectionId))
   }, [store])
 
-  useEffect(() => {
-    for (const credential of offers) {
-      if (credential.state == CredentialState.OfferReceived && credential.connectionId === workflowConnectionId) {
-        goToHomeScreen(credential.id)
-        deactivateSlider()
-      }
-    }
-  }, [offers, workflowConnectionId])
-
   useEffect(() => {
     const credentialDefinitionIDs = credentials.map(
       (c) => c.metadata.data[CredentialMetadataKeys.IndyCredential].credentialDefinitionId as string
     )
 
-    setShowGetFoundationCredential(showBCIDSelector(credentialDefinitionIDs, canUseLSBCredential))
-  }, [credentials, canUseLSBCredential])
+    setShowGetFoundationCredential(showBCIDSelector(credentialDefinitionIDs, true))
+  }, [credentials])
+
+  useEffect(() => {
+    const handle = DeviceEventEmitter.addListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, (value?: boolean) => {
+      const newVal = value === undefined ? !addCredentialPressed : value
+      setAddCredentialPressed(newVal)
+    })
+
+    return () => {
+      handle.remove()
+    }
+  }, [])
 
   useEffect(() => {
     const handle = DeviceEventEmitter.addListener(BCWalletEventTypes.ADD_CREDENTIAL_PRESSED, (value?: boolean) => {
```

Comment:
Maybe my eyes are bugging out but is this useEffect duplicated here? Also we could return just `handle.remove` as the cleanup function

Reply:
Merge conflicts...